rfc/rfc-0000-supply-chain-tracing.md

```diff
@@ -0,0 +1,182 @@
+    inputs:
+    - name:                  # string
+    outputs:
+    - name:                  # string
+      value:                 # json.RawExtension
+      lastTransitionTime:    # metav1.Time
+    observedGeneration:    # int64
+```
+
+Based on the [basic supply chain example](https://github.com/vmware-tanzu/cartographer/tree/53402edb8b8914b4cd36ace82da85f83f3daefc1/examples/basic-sc), the Workload status would have the form:
+
+```yaml
+... existing fields ...
+status:
+  ... existing fields ...
+  resources:
+  - name: source-provider
+    stampedRef:
+      apiVersion: source.toolkit.fluxcd.io/v1beta1
+      kind: GitRepository
+      namespace: default
+      name: my-workload
+    templateRef:
+      apiVersion: carto.run/v1
+      kind: ClusterSourceTemplate
+      name: source
+      outputs:
+      - name: url
+        value: http://source-controller.flux-system.svc.cluster.local./gitrepository/default/my-workload/3d42c19a618bb8fc13f72178b8b5e214a2f989c4.tar.gz
+        lastTransitionTime: "2022-02-16T03:29:52Z"
+      - name: revision
+        value: main/3d42c19a618bb8fc13f72178b8b5e214a2f989c4
+        lastTransitionTime: "2022-02-16T03:29:52Z"
+      observedGeneration: 1
+  - name: image-builder
+    stampedRef:
+      apiVersion: kpack.io/v1alpha2
+      kind: Image
+      namespace: default
+      name: my-workload
+    templateRef:
+      apiVersion: carto.run/v1
+      kind: ClusterImageTemplate
+      name: image
+    inputs:
+    - name: source-provider
+    outputs:
+    - name: image
+      value: registry.example/supply-chain/my-workload@sha256:68f8e8fc6e8ede7a411db9182cd695eac7b3e7e19e4ff9dcb9ba21205c135697
+      lastTransitionTime: "2022-02-16T03:23:37Z"
+      observedGeneration: 14
+  - name: deployer
+    stampedRef:
+      apiVersion: kappctrl.k14s.io/v1alpha1
+      kind: App
+      namespace: default
+      name: my-workload
+    templateRef:
+      apiVersion: carto.run/v1
+      kind: ClusterTemplate
+      name: app-deploy
+    inputs:
+    - name: image-builder
+    observedGeneration: 19
+```
+
+# How it Works
+[how-it-works]: #how-it-works
+
+- `.status.resources[*]` should match the order and length of `.spec.resource` for a ClusterSupplyChain.
+- `.status.resources[*].name` is the name of the resources as defined within the ClusterSupplyChain resource's array.
+- `.status.resources[*].templateRef` object reference to the template resource referenced by the ClusterSupplyChain.
+- `.status.resources[*].stampedRef` object reference to the Kubernetes resource created by Cartographer for this resource.
+- `.status.resources[*].inputs[*]` inputs model the relationships between resources within the SupplyChain graph. The value of an input can be read from the referenced resource's outputs.
+- `.status.resources[*].inputs[*].name` the name of the resource backing this input. 
+- `.status.resources[*].outputs[*].name` the name of the output. Output names are fixed and defined by the template type.
```

Comment:
(or perhaps even `resourceName` to really leave no room for other interpretations?)

Reply:
or ... what if we even removed the plural here (output**s**) and went solely with `output`? a resource can only export either one (`cluster(source|image|config)template`) or no output (`clustertemplate`)) 🤔

Reply:
or change `inputs` to `inputResources`?

I waffled a bit while drafting the RFCs whether `.inputs[*].name` should be `.inputs[*].resource` since the value is from the `resource` field within the ResourceReference struct. I ended up going with `name` here because it's the dominate part of the relationship and I was viewing this as a local object reference like resource.